cantusdb/turtle/csv2rdf_by_RDFLib.py

```diff
@@ -0,0 +1,71 @@
+from rdflib import Graph, URIRef, Literal
+from rdflib.namespace import RDF
+import csv
+import validators
+import sys
+
+# helper to get the predicate relation dictionary
+def get_ontology_dict () :
+    
+    # a specific config file for the predicates
+    # TODO: maybe change this part to make a more customized way of getting the predicates configurations
+    ontology_f = open("ontology_dict.csv", 'r')
+    ontology_reader = csv.DictReader(ontology_f)
+    ontology_dict = {}
+    
+    # converting the reader dict to the predicate dictionary
+    for row in ontology_reader:
+        ontology_dict[row[list(row.keys())[0]]] = row[list(row.keys())[1]]
+                
+    return ontology_dict
+    
+# a main operation function
+def convert_csv_to_turtle():
+    # TODO: maybe change this one to accomadate for multiple files
+    if len(sys.argv) != 2:
+        raise ValueError("Invalid number of input filename") 
+    
+    g = Graph()
+
+    csv_file = open(sys.argv[1], 'r', encoding='utf-8')
+    
+    csv_reader = csv.DictReader(csv_file)
+    ontology_dict = get_ontology_dict()
+
+    # Convert each row to Turtle format and add it to the output
+    for row in csv_reader:
+        # the first column as the subject
+        absolute_uri = URIRef(row[list(row.keys())[0]])
```

Comment:
`row[“absolute_url”]`?

Reply:
This script should be more general. It uses whatever is the first attribute of an entry in the input CSV file. In this case, it is the row["absolute_url"], but it can be other.

Reply:
If that's the case then you probably should be using a different CSV Reader class than the DictReader, since the ability to access columns by their header is the whole point of using the DictReader class.

Reply:
I will figure out this problem with the first conversation.